### PR TITLE
concurerency-utilites/IT-academy/M-JCI-55-20/Pashyk Yan/task done

### PR DESCRIPTION
**Pull request by stranger11**
## Pull request state

Please indicate actual status for PR:
- [x] on review
- [x] reviewed
- [ ] fixed
- [ ] on approve
- [ ] approved

### CODE REVIEW

**Comment by stranger11** [comment]
Hello, Dima, check please, this task, thank you!

File: src/main/java/com/rakovets/course/javabasics/practice/concurrency/utilities/ParallelCalculator.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                Integer sum = (Integer) result.get();
+                System.out.println("All elements sum array" + i + " : " + sum);
+            } catch (InterruptedException | ExecutionException e) {
+                e.printStackTrace();
+            }
+        }
+        executorService.shutdown();
+        long timeSpent = System.currentTimeMillis() - timeStart;
+        System.out.println(numberOfThreads + "threads time: " + timeSpent);
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/concurrency/utilities/ParallelCalculator.java:56]:
```suggestion
    public static void getCalculationInSeveralThreads(int numberOfThreads, List<Integer[]> list) {
        ExecutorService executorService = Executors.newFixedThreadPool(numberOfThreads);
        long timeStart = System.currentTimeMillis();
        Iterator<Integer[]> iter = list.iterator();
        int i = 0;

        List<Callable<Integer>> tasks = new LinkedList<>();

        while (iter.hasNext()) {
            Integer[] array = iter.next();
            Callable<Integer> task = new ParallelThread(array);
            tasks.add(task);
        }
        try {
            executorService.invokeAll(tasks);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        executorService.shutdown();
        long timeSpent = System.currentTimeMillis() - timeStart;
        System.out.println(numberOfThreads + " threads time: " + timeSpent);
    }
```

File: src/main/java/com/rakovets/course/javabasics/practice/concurrency/utilities/ParallelThreads.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        System.out.println("Length: " + array.length);
+        Integer sum = 0;
+        for(int item : array) {
+            sum +=  item;
+        } return sum;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/concurrency/utilities/ParallelThreads.java:19]:
```suggestion
    @Override
    public Integer call() throws Exception {
        System.out.println("Length: " + array.length);
        Integer sum = 0;
        for(int item : array) {
            sum +=  item;
        }
        Thread.sleep(200);
        System.out.println(Thread.currentThread().getName() + " - All elements sum array: " + sum);
        return sum;
    }
```

**Review by rakovets — COMMENTED** [review]
Not bad, but it works not correctly for multi threading.

**Pull request closed without merging**